### PR TITLE
docs(route53): hostedZone import functions pre-requisites and usage causing confusion

### LINKED ISSUE #5547: route53.HostedZone.fromLookup returned error
I have a hosted zone on route53, and use the following CDK code (`typescript`) to look up that zone using `domainName`, but got error while running `cdk synth`, `cdk diff` and `cdk deploy`. After changing to use `route53.HostedZone.fromHostedZoneAttributes()`, it works.

### Reproduction Steps

```
const zone = route53.HostedZone.fromLookup(this, 'Zone', { domainName: props.domainName });
```

### Error Log

```
Cannot retrieve value from context provider hosted-zone since account/region are not specified at the stack level. Either configure "env" with explicit account and region when you define your stack, or use the environment variables "CDK_DEFAULT_ACCOUNT" and "CDK_DEFAULT_REGION" to inherit environment information from the CLI (not recommended for production stacks)
Subprocess exited with error 1
```

### Environment

  - **CLI Version      : 1.19.0 (build 5597bbe)**
  - **Framework Version:1.19.0**
  - **OS               :macOS Mojave version 10.14.6**
  - **Language         :typescript **

### Other

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by BryanPan342**
Wrote up some documentation stuff but happy to change it to be more clear on the differences between `HostedZone.fromLookup`, `HostedZone.fromHostedZoneAttributes`, and `HostedZone.fromHostedZoneId`.

Fixes #5547 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-route53/README.md [hunk 1]
```diff
@@ -83,7 +83,19 @@ Constructs are available for A, AAAA, CAA, CNAME, MX, NS, SRV and TXT records.
...
 allowed to issue certificates for a domain to Amazon only.
 
-### Adding records to existing hosted zones
+### Imports
+
+If you don't know the ID of a Hosted Zone, you can use the `HostedZone.fromLookup`
+to discover and import it:
+
+Note that this requires an explicit environment to query an AWS account. Check
+out the [documentation](https://docs.aws.amazon.com/cdk/latest/guide/environments.html) for an overview of environments and how to implement.
```
Comment (shivlaks) [inline packages/@aws-cdk/aws-route53/README.md:92]: perhaps we should include an example ourselves showing basic usage

```suggestion
⚠️ `fromLookup` requires an environment to be configured. Check
out the [documentation](https://docs.aws.amazon.com/cdk/latest/guide/environments.html) for an overview and examples
```

File: packages/@aws-cdk/aws-route53/README.md [hunk 2]
```diff
@@ -83,7 +83,19 @@ Constructs are available for A, AAAA, CAA, CNAME, MX, NS, SRV and TXT records.
 Use the `CaaAmazonRecord` construct to easily restrict certificate authorities
 allowed to issue certificates for a domain to Amazon only.
 
-### Adding records to existing hosted zones
+### Imports
+
```
Comment (shivlaks) [inline packages/@aws-cdk/aws-route53/README.md:87]:
```suggestion

Any Route53 Hosted Zone that has been created outside the stack can be imported into your CDK app
```

File: packages/@aws-cdk/aws-route53/README.md [hunk 3]
```diff
@@ -83,7 +83,19 @@ Constructs are available for A, AAAA, CAA, CNAME, MX, NS, SRV and TXT records.
...
+If you don't know the ID of a Hosted Zone, you can use the `HostedZone.fromLookup`
```
Comment (shivlaks) [inline packages/@aws-cdk/aws-route53/README.md:88]:
```suggestion
If you don't know the ID of the Hosted Zone to import, you can use the `HostedZone.fromLookup`
```

File: packages/@aws-cdk/aws-route53/lib/hosted-zone.ts [hunk 4]
```diff
@@ -59,6 +59,8 @@ export class HostedZone extends Resource implements IHostedZone {
   /**
    * Import a Route 53 hosted zone defined either outside the CDK, or in a different CDK stack
    *
+   * Use when hosted zone ID is known and hosted zone name is undesirable.
```
Comment (shivlaks) [inline packages/@aws-cdk/aws-route53/lib/hosted-zone.ts:62]: maybe reword to say that the zone name will be unavailable if this API is used for import

File: packages/@aws-cdk/aws-route53/lib/hosted-zone.ts [hunk 5]
```diff
@@ -94,6 +102,11 @@ export class HostedZone extends Resource implements IHostedZone {
 
   /**
    * Lookup a hosted zone in the current account/region based on query parameters.
+   * Requires explicit environment, you must specify env in the stack.
```
Comment (shivlaks) [inline packages/@aws-cdk/aws-route53/lib/hosted-zone.ts:105]:
```suggestion
   * Requires environment, you must specify env for the stack.
```

File: packages/@aws-cdk/aws-route53/lib/hosted-zone.ts [hunk 6]
```diff
@@ -94,6 +102,11 @@ export class HostedZone extends Resource implements IHostedZone {
...
+   *
+   * Use to easily query hosted zones.
+   *
+   * Look at README for more documentation.
```
Comment (shivlaks) [inline packages/@aws-cdk/aws-route53/lib/hosted-zone.ts:109]:
```suggestion
   * @see https://docs.aws.amazon.com/cdk/latest/guide/environments.html
```

File: packages/@aws-cdk/aws-route53/README.md (hunk 1 above)
Comment (BryanPan342) [reply packages/@aws-cdk/aws-route53/README.md:92]: yeah i think this is a good idea! i think something simple like 

```ts
new MyDevStack(app, 'dev', { 
  env: { 
    account: process.env.CDK_DEFAULT_ACCOUNT, 
    region: process.env.CDK_DEFAULT_REGION 
}});
```
and some description of using profiles should be sufficient, what do you think?
